packaging/microshift/config.yaml

```diff
@@ -41,3 +41,9 @@ cluster:
 
 # The name of the node (defaults to hostname)
 #nodeName: ""
+
+# OVNKubernetes network plugin config
+#ovnkubernetesConfig:
```

Comment:
We found a way to have all of the configuration for ODF-LVM in a file outside of MicroShift's main config file. That will, eventually, allow us to provide a different CSI implementation that can also have its own separate file.

Some of the future use cases for MicroShift, especially upstream, make it desirable to be able to replace the CNI implementation as well, and following the same pattern with a similar clean separation for the configuration will make that more straightforward. 

Let's talk about the approach we want to take this week.

Reply:
@dhellmann ack+, I updated the PR to follow the CSI approach and have a separate config file for ovn-kubernetes.

Reply:
That looks good.

I'm not entirely comfortable with parsing YAML from bash, but what you have looks like it will work and we can consider whether we need something more "robust" in the future if we run into problems.

Reply:
I thought about using `yq` in the bash to parse yaml file, but it's not available in the existing repos. Adding the installation dependency sounds a bit heavy for this change in configure-ovs.sh since it is currently the only one I'm aware of that requires parsing yaml in bash in microshift project.
Another idea might be to use json config file and `jq` to parse it, `jq` is installed by default. The downside is that it is inconsistent with csi and microshift config files. I could write a small golang program to parse the yaml if we will have similar needs in microshift scripts. Or any other ideas?

Reply:
Like I said, I think what you have is fine. We can consider other options as an improvement later if we have issues or establish a pattern based on other config.